File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> m;
    for(auto p: participant) m[p]++;
    for(auto c: completion) {
        if(m.find(c) == m.end() || m[c] == 0) return c;
        m[c]--;
    }
    for(auto p: participant) if(m[p] != 0) return p;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> m = new HashMap<>();
        
        for(String cur : participant) m.put(cur, m.getOrDefault(cur, 0) + 1);
        for(String cur : completion) m.put(cur, m.get(cur) - 1);
        
        for(Map.Entry<String,Integer> entry : m.entrySet()){
            if(entry.getValue() == 1) return entry.getKey();
        }
       
        return "";
    }
}